drivers/adc/adc_mchp_xec_v2.c

```diff
@@ -304,45 +307,66 @@ static int adc_xec_init(const struct device *dev)
 	const struct adc_xec_config *const cfg = ADC_XEC_CONFIG(dev);
 	struct adc_xec_regs *adc_regs = ADC_XEC_REG_BASE(dev);
 	struct adc_xec_data *data = ADC_XEC_DATA(dev);
+	int ret;
+
+	ret = pinctrl_apply_state(cfg->pcfg, PINCTRL_STATE_DEFAULT);
+	if (ret != 0) {
+		LOG_ERR("XEC ADC V2 pinctrl setup failed (%d)", ret);
+		return ret;
+	}
 
 	adc_regs->control_reg =  XEC_ADC_CTRL_ACTIVATE
 		| XEC_ADC_CTRL_POWER_SAVER_DIS
 		| XEC_ADC_CTRL_SINGLE_DONE_STATUS
 		| XEC_ADC_CTRL_REPEAT_DONE_STATUS;
 
+	cfg->irq_config_func();
+
 	mchp_xec_ecia_girq_src_dis(cfg->girq_single, cfg->girq_single_pos);
 	mchp_xec_ecia_girq_src_dis(cfg->girq_repeat, cfg->girq_repeat_pos);
 	mchp_xec_ecia_girq_src_clr(cfg->girq_single, cfg->girq_single_pos);
 	mchp_xec_ecia_girq_src_clr(cfg->girq_repeat, cfg->girq_repeat_pos);
 	mchp_xec_ecia_girq_src_en(cfg->girq_single, cfg->girq_single_pos);
 
-	IRQ_CONNECT(DT_INST_IRQN(0),
-		    DT_INST_IRQ(0, priority),
-		    adc_xec_isr, DEVICE_DT_INST_GET(0), 0);
-	irq_enable(DT_INST_IRQN(0));
-
 	adc_context_unlock_unconditionally(&data->ctx);
 
 	return 0;
 }
 
-static struct adc_xec_config adc_xec_dev_cfg_0 = {
-	.base_addr = (uintptr_t)(DT_INST_REG_ADDR(0)),
-	.girq_single = (uint8_t)(DT_INST_PROP_BY_IDX(0, girqs, 0)),
-	.girq_single_pos = (uint8_t)(DT_INST_PROP_BY_IDX(0, girqs, 1)),
-	.girq_repeat = (uint8_t)(DT_INST_PROP_BY_IDX(0, girqs, 2)),
-	.girq_repeat_pos = (uint8_t)(DT_INST_PROP_BY_IDX(0, girqs, 3)),
-	.pcr_regidx = (uint8_t)(DT_INST_PROP_BY_IDX(0, pcrs, 0)),
-	.pcr_bitpos = (uint8_t)(DT_INST_PROP_BY_IDX(0, pcrs, 1)),
-};
-
-static struct adc_xec_data adc_xec_dev_data_0 = {
-	ADC_CONTEXT_INIT_TIMER(adc_xec_dev_data_0, ctx),
-	ADC_CONTEXT_INIT_LOCK(adc_xec_dev_data_0, ctx),
-	ADC_CONTEXT_INIT_SYNC(adc_xec_dev_data_0, ctx),
-};
-
-DEVICE_DT_INST_DEFINE(0, adc_xec_init, NULL,
-		    &adc_xec_dev_data_0, &adc_xec_dev_cfg_0,
-		    PRE_KERNEL_1, CONFIG_ADC_INIT_PRIORITY,
-		    &adc_xec_api);
+#define ADC_XEC_DEVICE(i)						\
```

Comment:
another example: converting drivers to be multi-instance is ok, but this needs to have its own commit